Possibility to disable kafka-streams if no Stream is defined

### Issue description

Versions:
- micronaut: 3.10.0
- micronaut-kafka-streams: 4.5.4

I am currently running into the issue that as soon as I pull in the micronaut-kafka-streams dependency AND I don't define any Streams that the server doesn't start up. It fails with:

```
Message: Invalid topology: Topology has no stream threads and no global threads, must subscribe to at least one source topic or global table.
Path Taken: KafkaStreams.kafkaStreams(String name,ConfiguredStreamBuilder builder,KafkaClientSupplier kafkaClientSupplier,KStream[] kStreams)
io.micronaut.context.exceptions.BeanInstantiationException: Bean definition [org.apache.kafka.streams.KafkaStreams] could not be loaded: Error instantiating bean of type  [org.apache.kafka.streams.KafkaStreams]

Message: Invalid topology: Topology has no stream threads and no global threads, must subscribe to at least one source topic or global table.
Path Taken: KafkaStreams.kafkaStreams(String name,ConfiguredStreamBuilder builder,KafkaClientSupplier kafkaClientSupplier,KStream[] kStreams)
        at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1921)
        at io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:249)
        at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:3326)
        at io.micronaut.context.DefaultBeanContext.finalizeConfiguration(DefaultBeanContext.java:3684)
        at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:341)
        at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:194)
        at io.micronaut.runtime.Micronaut.start(Micronaut.java:75)
        at op.events.ApplicationKt.main(Application.kt:10)
Caused by: io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type  [org.apache.kafka.streams.KafkaStreams]
```

I assume this comes from [here](https://github.com/micronaut-projects/micronaut-kafka/blob/master/kafka-streams/src/main/java/io/micronaut/configuration/kafka/streams/KafkaStreamsFactory.java#L113C9-L113C21)  because `org.apache.kafka.streams.KafkaStreams` requires at least 1 Stream from my understanding.

I am aware of the `kafka.streams.default.start-kafka-streams` setting, however this only influences whether Kafka is started or not. With this configuration set to false, it is still required to have at list one Stream Bean defined.

Is this intended or is there a nice way to overcome this problem, right now we have something like this to avoid the startup problem:

```
@Factory
class MicronautKafkaStreamsDisabler {

    @Singleton
    @Requires(property = START_KAFKA_STREAMS_CONFIG, value = "false", defaultValue = "true")
    fun noOpStream(builder: ConfiguredStreamBuilder): KStream<String, ByteArray> =
        builder.stream("", Consumed.with(Serdes.String(), Serdes.ByteArray()))
}
```

By setting `kafka.streams.default.start-kafka-streams: false` this noOpStream Bean is created.

Use case: We have a project that has Kafka Consumers and Kafka Streams which are contained in a single docker image. On deployment, we want to run Consumers and Streams separately in different pods. So the pod that is running a consumer only will not have a Stream bean enabled.

--- a/kafka-streams/src/main/java/io/micronaut/configuration/kafka/streams/DefaultKafkaStreamsConfiguration.java
+++ b/kafka-streams/src/main/java/io/micronaut/configuration/kafka/streams/DefaultKafkaStreamsConfiguration.java
@@ -19,6 +19,7 @@ import io.micronaut.configuration.kafka.config.KafkaDefaultConfiguration;
 import io.micronaut.context.annotation.Primary;
 import io.micronaut.context.annotation.Requires;
 import io.micronaut.context.env.Environment;
+import io.micronaut.core.util.StringUtils;
 import io.micronaut.runtime.ApplicationConfiguration;
 import jakarta.inject.Named;
 import jakarta.inject.Singleton;
@@ -36,6 +37,7 @@ import java.util.Properties;
 @Requires(missingProperty = KafkaStreamsConfiguration.PREFIX + ".default")
 @Singleton
 @Requires(beans = KafkaDefaultConfiguration.class)
+@Requires(property = KafkaStreamsConfiguration.ENABLED, notEquals = StringUtils.FALSE, defaultValue = StringUtils.TRUE)
 @Named(AbstractKafkaStreamsConfiguration.DEFAULT_NAME)
 @Primary
 public class DefaultKafkaStreamsConfiguration<K, V> extends AbstractKafkaStreamsConfiguration<K, V> {

--- a/kafka-streams/src/main/java/io/micronaut/configuration/kafka/streams/KafkaStreamsConfiguration.java
+++ b/kafka-streams/src/main/java/io/micronaut/configuration/kafka/streams/KafkaStreamsConfiguration.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2017-2020 original authors
+ * Copyright 2017-2026 original authors
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -20,6 +20,7 @@ import io.micronaut.context.annotation.ConfigurationProperties;
 import io.micronaut.context.annotation.EachProperty;
 import io.micronaut.context.annotation.Parameter;
 import io.micronaut.context.annotation.Requires;
+import io.micronaut.context.exceptions.DisabledBeanException;
 import io.micronaut.context.env.Environment;
 import io.micronaut.core.naming.NameUtils;
 import io.micronaut.core.util.StringUtils;
@@ -65,6 +66,9 @@ public class KafkaStreamsConfiguration<K, V> extends AbstractKafkaStreamsConfigu
             ApplicationConfiguration applicationConfiguration,
             Environment environment) {
         super(defaultConfiguration);
+        if ("enabled".equals(streamName)) {
+            throw new DisabledBeanException("Global property " + ENABLED + " is not a stream configuration");
+        }
         setName(streamName);
         Properties config = getConfig();
         String propertyKey = PREFIX + '.' + NameUtils.hyphenate(streamName, true);

--- a/kafka-streams/src/main/java/io/micronaut/configuration/kafka/streams/KafkaStreamsConfiguration.java
+++ b/kafka-streams/src/main/java/io/micronaut/configuration/kafka/streams/KafkaStreamsConfiguration.java
@@ -22,6 +22,7 @@ import io.micronaut.context.annotation.Parameter;
 import io.micronaut.context.annotation.Requires;
 import io.micronaut.context.env.Environment;
 import io.micronaut.core.naming.NameUtils;
+import io.micronaut.core.util.StringUtils;
 import io.micronaut.runtime.ApplicationConfiguration;
 
 import java.util.Properties;
@@ -37,12 +38,18 @@ import static io.micronaut.configuration.kafka.streams.KafkaStreamsConfiguration
 @EachProperty(value = PREFIX, primary = "default")
 @ConfigurationProperties(PREFIX)
 @Requires(beans = KafkaDefaultConfiguration.class)
+@Requires(property = KafkaStreamsConfiguration.ENABLED, notEquals = StringUtils.FALSE, defaultValue = StringUtils.TRUE)
 public class KafkaStreamsConfiguration<K, V> extends AbstractKafkaStreamsConfiguration<K, V> {
 
     /**
      * The default streams configuration.
      */
     public static final String PREFIX = "kafka.streams";
+
+    /**
+     * Global property used to disable Kafka Streams bean creation.
+     */
+    public static final String ENABLED = PREFIX + ".enabled";
 
     /**
      * Construct a new {@link KafkaStreamsConfiguration} for the given defaults.

--- a/kafka-streams/src/main/java/io/micronaut/configuration/kafka/streams/KafkaStreamsFactory.java
+++ b/kafka-streams/src/main/java/io/micronaut/configuration/kafka/streams/KafkaStreamsFactory.java
@@ -21,6 +21,7 @@ import io.micronaut.context.BeanProvider;
 import io.micronaut.context.annotation.*;
 import io.micronaut.context.exceptions.DisabledBeanException;
 import io.micronaut.context.event.ApplicationEventPublisher;
+import io.micronaut.core.util.StringUtils;
 import jakarta.annotation.PreDestroy;
 import jakarta.inject.Singleton;
 import org.apache.kafka.streams.KafkaClientSupplier;
@@ -52,6 +53,7 @@ import static java.util.function.Predicate.not;
  * @since 1.0
  */
 @Factory
+@Requires(property = KafkaStreamsConfiguration.ENABLED, notEquals = StringUtils.FALSE, defaultValue = StringUtils.TRUE)
 public class KafkaStreamsFactory implements Closeable {
 
     private static final Logger LOG = LoggerFactory.getLogger(KafkaStreamsFactory.class);

--- a/kafka-streams/src/main/java/io/micronaut/configuration/kafka/streams/health/KafkaStreamsHealth.java
+++ b/kafka-streams/src/main/java/io/micronaut/configuration/kafka/streams/health/KafkaStreamsHealth.java
@@ -53,6 +53,7 @@ import java.util.stream.Stream;
  */
 @Singleton
 @Requires(classes = HealthIndicator.class)
+@Requires(beans = KafkaStreamsFactory.class)
 @Requires(property = KafkaStreamsHealth.ENABLED_PROPERTY, value = "true", defaultValue = "true")
 public class KafkaStreamsHealth implements HealthIndicator {
 

--- a/kafka-streams/src/test/groovy/io/micronaut/configuration/kafka/streams/GlobalKTableOnlySpec.groovy
+++ b/kafka-streams/src/test/groovy/io/micronaut/configuration/kafka/streams/GlobalKTableOnlySpec.groovy
@@ -37,6 +37,31 @@ class GlobalKTableOnlySpec extends AbstractKafkaSpec {
         topologyDescription.contains(GlobalTableOnlyFactory.STORE)
     }
 
+    void "global kafka streams enabled property is not treated as a stream name"() {
+        given:
+        ApplicationContext enabledContext = ApplicationContext.run([
+                'spec.name'                                           : 'GlobalKTableOnlySpec',
+                'kafka.test.initializer.enabled'                      : false,
+                'kafka.bootstrap.servers'                             : 'localhost:9092',
+                'kafka.streams.enabled'                               : true,
+                'kafka.streams.default.application.id'                : 'default-enabled-' + UNIQUE_SUFFIX,
+                'kafka.streams.default.start-kafka-streams'           : false,
+                'kafka.streams.default.state.dir'                     : TMP_DIR + '/enabled-property-default-' + UNIQUE_SUFFIX,
+                'kafka.streams.global-table-only.application.id'      : 'global-table-only-' + UNIQUE_SUFFIX,
+                'kafka.streams.global-table-only.start-kafka-streams' : false,
+                'kafka.streams.global-table-only.state.dir'           : TMP_DIR + '/enabled-property-global-' + UNIQUE_SUFFIX
+        ])
+
+        when:
+        List<String> streamNames = enabledContext.getBeansOfType(ConfiguredStreamBuilder)*.name.sort()
+
+        then:
+        !streamNames.contains('enabled')
+
+        cleanup:
+        enabledContext.close()
+    }
+
     @Override
     protected Map<String, Object> getConfiguration() {
         super.getConfiguration() + [

--- a/kafka-streams/src/test/groovy/io/micronaut/configuration/kafka/streams/KafkaStreamsDisabledSpec.groovy
+++ b/kafka-streams/src/test/groovy/io/micronaut/configuration/kafka/streams/KafkaStreamsDisabledSpec.groovy
@@ -1,0 +1,24 @@
+package io.micronaut.configuration.kafka.streams
+
+import io.micronaut.configuration.kafka.streams.health.KafkaStreamsHealth
+import io.micronaut.context.ApplicationContext
+import spock.lang.AutoCleanup
+
+class KafkaStreamsDisabledSpec extends AbstractKafkaSpec {
+
+    @AutoCleanup
+    ApplicationContext context = ApplicationContext.run([
+            'spec.name'              : 'KafkaStreamsDisabledSpec',
+            'kafka.test.initializer.enabled': false,
+            'kafka.bootstrap.servers': 'localhost:9092',
+            'kafka.streams.enabled'  : false
+    ])
+
+    void "global kafka streams disable removes streams beans"() {
+        expect:
+        !context.findBean(DefaultKafkaStreamsConfiguration).present
+        !context.findBean(ConfiguredStreamBuilder).present
+        !context.findBean(KafkaStreamsFactory).present
+        !context.findBean(KafkaStreamsHealth).present
+    }
+}

--- a/src/main/docs/guide/kafkaStreams.adoc
+++ b/src/main/docs/guide/kafkaStreams.adoc
@@ -126,6 +126,10 @@ NOTE: The `@Named` qualifier must be applied to the injected `ConfiguredStreamBu
 
 snippet::io.micronaut.kafka.docs.streams.WordCountStream[tags="namedStream", indent=0]
 
+.Disabling Kafka Streams integration
+
+If an application includes the `kafka-streams` module but should not initialize any Kafka Streams beans in a given environment, set `kafka.streams.enabled` to `false`.
+
 .Configuring Kafka Streams for testing
 When writing a test without starting the actual Kafka server, you can instruct Micronaut not to start Kafka Streams. To do this create a config file suffixed with an environment name, such as `application-test.yml` and set the `kafka.streams.[STREAM-NAME].start-kafka-streams` to `false`.
 

--- a/test-suite/src/test/java/io/micronaut/kafka/docs/MyTest.java
+++ b/test-suite/src/test/java/io/micronaut/kafka/docs/MyTest.java
@@ -12,10 +12,11 @@ import io.micronaut.testcontainers.kafka.Kafka;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.TestInstance;
 
+import java.util.concurrent.LinkedBlockingQueue;
 import java.util.Map;
+import java.util.concurrent.TimeUnit;
 
-import static java.util.concurrent.TimeUnit.SECONDS;
-import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertEquals;
 
 @Property(name = "spec.name", value = "MyTest")
 @MicronautTest
@@ -28,10 +29,10 @@ class MyTest implements TestPropertyProvider {
     }
 
     @Test
-    void testKafkaRunning(MyProducer producer, MyConsumer consumer) {
+    void testKafkaRunning(MyProducer producer, MyConsumer consumer) throws InterruptedException {
         final String message = "hello";
         producer.produce(message);
-        await().atMost(5, SECONDS).until(() -> message.equals(consumer.consumed));
+        assertEquals(message, consumer.awaitMessage(15, TimeUnit.SECONDS));
     }
 
     @Requires(property = "spec.name", value = "MyTest")
@@ -44,10 +45,15 @@ class MyTest implements TestPropertyProvider {
     @Requires(property = "spec.name", value = "MyTest")
     @KafkaListener(offsetReset = OffsetReset.EARLIEST)
     static class MyConsumer {
-        String consumed;
+        private final LinkedBlockingQueue<String> consumedMessages = new LinkedBlockingQueue<>();
+
         @Topic("my-topic")
         public void consume(String message) {
-            consumed = message;
+            consumedMessages.offer(message);
+        }
+
+        String awaitMessage(long timeout, TimeUnit timeUnit) throws InterruptedException {
+            return consumedMessages.poll(timeout, timeUnit);
         }
     }
 }

